server/src/main/java/org/opensearch/indices/AbstractIndexShardCacheEntity.java

```diff
@@ -66,7 +67,19 @@ public final void onMiss() {
     }
 
     @Override
-    public final void onRemoval(RemovalNotification<IndicesRequestCache.Key, BytesReference> notification) {
-        stats().onRemoval(notification.getKey(), notification.getValue(), notification.getRemovalReason() == RemovalReason.EVICTED);
+    public final void onRemoval(RemovalNotification<ICacheKey<IndicesRequestCache.Key>, BytesReference> notification) {
+        stats().onRemoval(
+            getRamBytesUsedInKey(notification.getKey()),
+            notification.getValue(),
+            notification.getRemovalReason() == RemovalReason.EVICTED
+        );
+    }
+
+    private long getRamBytesUsedInKey(ICacheKey<IndicesRequestCache.Key> key) {
+        long innerKeyRamBytesUsed = 0;
+        if (key.key != null) {
+            innerKeyRamBytesUsed = key.key.ramBytesUsed();
+        }
+        return key.ramBytesUsed(innerKeyRamBytesUsed);
```

Comment:
Is this setter within the Key class? If yes, where is it being used for reporting ramBytesUsed?

Reply:
ICacheKey.ramBytesUsed() isn't a setter, it just takes in the size of the inner key (key.key, in this case IndicesRequestCache.Key which is Accountable) and returns the total size of the ICacheKey, including dimensions. 

Ideally we could enforce K in ICacheKey<K> is Accountable. Then ICacheKey could also be Accountable, we could just define ICacheKey.ramBytesUsed() with no arguments, and pass this value to ShardRequestCache. But we can't do this for the same reason we couldn't enforce K to implement equals(), as String isn't Accountable and we want to be able to use ICacheKey<String>.

Reply:
Considering we are only using ICacheKey<String> in our tests for now, why can't we wrap that inside something like `AccountableString` and use it accordingly. And then we can enforce K to be Accountable?
This approach won't require these changes, what do you think?